shared-bindings/nvm/ByteArray.h

```diff
@@ -31,13 +31,13 @@
 
 extern const mp_obj_type_t nvm_bytearray_type;
 
-uint32_t common_hal_nvm_bytearray_get_length(nvm_bytearray_obj_t *self);
+uint32_t common_hal_nvm_bytearray_get_length(const nvm_bytearray_obj_t *self);
```

Comment:
@tannewt specifying `const` for the nvm object builds and works in my cursory tests, allowing me to use `common_hal_mcu_nvm_obj` rather than creating my own. Just not sure if this has longer reaching implications?